### ISSUE TITLE
17.2.0 release tag is not reachable from 'master'

### ISSUE THREAD

**Issue by warner** [pyca/pyopenssl#680]
Most of the release tags are ancestors of the current 'master' commit, but 17.2.0 is not (it was produced on a short side-branch, which was never merged back into trunk). In addition, there are no branches which point to the 17.2.0 tag's commit: the tag is the only thing keeping that revision alive.

As a result, if you had a git checkout from before July 20th, and do a `git pull`, you'll get the new `master` but you won't ever see the 17.2.0 release.

I think that, if a release can't be made on trunk (for whatever reason), then there should be a merge from that release back into master (even if it is a "keep all my current code, ignore their side" -style pseudo-merge), so that all meaningful commits and releases will be ancestors of the current trunk.

If people agree, I'll make up a PR that basically just inherits from the 17.2.0 tag and current `master` (23c965e) without making any code changes.

**Comment by reaperhulk** [comment]
Looks like maybe a mistake was made during release and 17.2.0 was tagged off the original commit and not the squash commit to master? @hynek?

**Comment by hynek** [comment]
`¯\_(ツ)_/¯`

How do we fix it?

**Comment by hynek** [comment]
I *think* I have fixed it?

**Comment by warner** [comment]
I did a fresh clone, and I now see a 17.2.0 release tag on commit dd44662 (which is indeed an ancestor of 'master' at 23c965e). In my earlier clone (when I filed this ticket 10 hours ago), the 17.2.0 release tag was on commit 8ca87e0. The code trees for the old and new 17.2.0 tag appear to be identical.

If y'all are ok with changing what a release tag points to, then yeah, this is now fixed. Someone who updates an old repo might see the tag being changed, or git might complain about it, or they might stick with the old value.. I'm not sure what git does in that case. But a new clone, and a pull in a repo that's too old to have seen the original 17.2.0 tag, should both get the right thing.

(if you want to be sticklers about tags being immutable, then you'd probably need to move the tag back to 8ca87e0 and do the kind of merge commit that I described to make sure that trunk is a descendant of the old tag. But personally I'm ok with the current situation).
